controllers/clusterversion_controller.go

```diff
@@ -203,6 +206,20 @@ func (c *ClusterVersionReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
+	sideCarConfig, err := c.getSideCarConfig()
```

Comment:
I don't think you need a func for this and also doing the error handling twice w/o any benefit.